test/tests/time_steppers/iteration_adaptive/gold/adapt_tstep_grow_dtfunc_restart_out.csv

```diff
@@ -0,0 +1,4 @@
+time,_dt
+5,2
+10,5
```

Comment:
This test had to be regolded (for which I just converted to a csv since the exodus was overkill for a timestepper test). Ultimately, this is was not producing the correct results (dt=5 after sync) but now matches [adapt_tstep_grow_dtfunc_out.csv](https://github.com/idaholab/moose/pull/20304/files#diff-cdaec0366328c8b80a3d2f979319d7b6b06a2753705eb7f73113f64056069808)